crates/solver/src/liquidity/order_converter.rs

```diff
@@ -75,10 +100,10 @@ impl SettlementHandling<LimitOrder> for OrderSettlementHandler {
     fn encode(&self, executed_amount: U256, encoder: &mut SettlementEncoder) -> Result<()> {
         let is_native_token_buy_order = self.order.data.buy_token == BUY_ETH_ADDRESS;
 
-        if self.order.metadata.class != OrderClass::Liquidity && is_native_token_buy_order {
-            // liquidity orders don't need an additional token equivalency, as the buy tokens
-            // clearing prices are not stored in the clearing prices vector, but in the
-            // LiquidityOrderTrade
+        if self.order.metadata.class == OrderClass::Ordinary && is_native_token_buy_order {
```

Comment:
Naming-nit: Unrelated to this PR, but I think `market` would be a better name for the "ordinary" orders, which are CoW Protocol market orders that are expected to settle right away.